src/models/serialportctrl.ts

```diff
@@ -20,70 +18,36 @@ export class SerialPortCtrl {
     return os.platform();
   }
 
-  static getComList(): Promise<SerialPortInfo[]> {
+  static async getComList() {
     return new Promise(
       (
-        resolve: (value: SerialPortInfo[]) => void,
+        resolve: (value: Object) => void,
         reject: (error: Error) => void
       ) => {
         // tslint:disable-next-line: no-any
-        serialport.list((err: any, ports: SerialPortInfo[]) => {
+        var portList: Array<Object> = [];
+        SerialPort.list((err: any, ports: SerialPortInfo[]) => {
           if (err) {
             reject(err);
           } else {
-            resolve(ports);
-          }
-        });
-      }
-    );
-  }
-
-  static async chooseCOM(board: Board | undefined): Promise<string> {
-    return new Promise(
-      async (
-        resolve: (value: string) => void,
-        reject: (reason: Error) => void
-      ) => {
-        let comList;
-        try{
-          comList = await SerialPortCtrl.getComList();
-        } catch (err) {
-          return reject(err);
-        }
-
-        const az3166 = board;
-
-        if (!az3166) {
-          return reject(new Error('AZ3166 is not found in the board list.'));
-        }
-
-        const list = _.filter(comList, com => {
-          if (com.vendorId && com.productId && az3166.vendorId &&
-            az3166.productId &&
-            com.vendorId.toLowerCase().endsWith(az3166.vendorId) &&
-            com.productId.toLowerCase().endsWith(az3166.productId)) {
-            return true;
-          } else {
-            return false;
+            ports.forEach( async (port) => {
+              const com = {
+                "comName": port.comName,
+                "productId": port.productId,
+                "vendorId": port.vendorId
+              };
+              try {
+                await portList.push(com);

```

Comment:
Why do we need `await` here? `push` should not be `async` method

Reply:
Fixed.